### ISSUE TITLE
Running in Dev Container on Apple M1 fails: ERROR [dev_container_auto_added_stage_label 4/4] RUN bash ./setup.sh

### ISSUE THREAD

**Issue by ChristianWeyer** [deislabs/spiderlightning#402]
**Description of the bug**

Running in Dev Container:

```
 => ERROR [dev_container_auto_added_stage_label 4/4] RUN bash ./setup.sh   8.1s 
------                                                                          
 > [dev_container_auto_added_stage_label 4/4] RUN bash ./setup.sh:              
#0 0.216 Get:1 http://ports.ubuntu.com/ubuntu-ports jammy InRelease [270 kB]
#0 0.364 Get:2 http://ports.ubuntu.com/ubuntu-ports jammy-updates InRelease [119 kB]
#0 0.397 Get:3 http://ports.ubuntu.com/ubuntu-ports jammy-backports InRelease [108 kB]
#0 0.433 Get:4 http://ports.ubuntu.com/ubuntu-ports jammy-security InRelease [110 kB]
#0 0.469 Get:5 http://ports.ubuntu.com/ubuntu-ports jammy/universe arm64 Packages [17.2 MB]
#0 2.024 Get:6 http://ports.ubuntu.com/ubuntu-ports jammy/main arm64 Packages [1758 kB]
#0 2.094 Get:7 http://ports.ubuntu.com/ubuntu-ports jammy/restricted arm64 Packages [24.2 kB]
#0 2.096 Get:8 http://ports.ubuntu.com/ubuntu-ports jammy/multiverse arm64 Packages [224 kB]
#0 2.115 Get:9 http://ports.ubuntu.com/ubuntu-ports jammy-updates/restricted arm64 Packages [180 kB]
#0 2.130 Get:10 http://ports.ubuntu.com/ubuntu-ports jammy-updates/main arm64 Packages [739 kB]
#0 2.218 Get:11 http://ports.ubuntu.com/ubuntu-ports jammy-updates/multiverse arm64 Packages [22.9 kB]
#0 2.276 Get:12 http://ports.ubuntu.com/ubuntu-ports jammy-updates/universe arm64 Packages [1033 kB]
#0 2.499 Get:13 http://ports.ubuntu.com/ubuntu-ports jammy-backports/main arm64 Packages [49.0 kB]
#0 2.502 Get:14 http://ports.ubuntu.com/ubuntu-ports jammy-backports/universe arm64 Packages [25.1 kB]
#0 2.506 Get:15 http://ports.ubuntu.com/ubuntu-ports jammy-security/restricted arm64 Packages [179 kB]
#0 2.521 Get:16 http://ports.ubuntu.com/ubuntu-ports jammy-security/main arm64 Packages [461 kB]
#0 2.603 Get:17 http://ports.ubuntu.com/ubuntu-ports jammy-security/universe arm64 Packages [800 kB]
#0 2.630 Get:18 http://ports.ubuntu.com/ubuntu-ports jammy-security/multiverse arm64 Packages [18.7 kB]
#0 2.727 Fetched 23.3 MB in 3s (9049 kB/s)
#0 2.727 Reading package lists...
#0 3.131 Reading package lists...
#0 3.509 Building dependency tree...
#0 3.584 Reading state information...
#0 3.594 E: Unable to locate package  
#0 3.596 ./setup.sh: line 16: unzip: command not found
#0 3.619 Reading package lists...
#0 3.994 Building dependency tree...
#0 4.087 Reading state information...
#0 4.170 The following additional packages will be installed:
#0 4.170   bzip2 libdpkg-perl libfile-fcntllock-perl libgdbm-compat4 libgdbm6
#0 4.170   libglib2.0-0 libglib2.0-data libicu70 liblocale-gettext-perl libperl5.34
#0 4.170   libxml2 netbase perl perl-modules-5.34 shared-mime-info xdg-user-dirs
#0 4.170   xz-utils
#0 4.170 Suggested packages:
#0 4.170   bzip2-doc debian-keyring gnupg gcc | c-compiler binutils patch git bzr
#0 4.170   gdbm-l10n perl-doc libterm-readline-gnu-perl | libterm-readline-perl-perl
#0 4.170   make libtap-harness-archive-perl dpkg-dev
#0 4.195 The following NEW packages will be installed:
#0 4.195   bzip2 libdpkg-perl libfile-fcntllock-perl libgdbm-compat4 libgdbm6
#0 4.195   libglib2.0-0 libglib2.0-data libicu70 liblocale-gettext-perl libperl5.34
#0 4.195   libxml2 netbase perl perl-modules-5.34 pkg-config shared-mime-info
#0 4.195   xdg-user-dirs xz-utils
#0 4.263 0 upgraded, 18 newly installed, 0 to remove and 10 not upgraded.
#0 4.263 Need to get 21.6 MB of archives.
#0 4.263 After this operation, 95.7 MB of additional disk space will be used.
#0 4.263 Get:1 http://ports.ubuntu.com/ubuntu-ports jammy/main arm64 liblocale-gettext-perl arm64 1.07-4build3 [16.9 kB]
#0 4.318 Get:2 http://ports.ubuntu.com/ubuntu-ports jammy-updates/main arm64 perl-modules-5.34 all 5.34.0-3ubuntu1.1 [2976 kB]
#0 4.722 Get:3 http://ports.ubuntu.com/ubuntu-ports jammy/main arm64 libgdbm6 arm64 1.23-1 [34.1 kB]
#0 4.723 Get:4 http://ports.ubuntu.com/ubuntu-ports jammy/main arm64 libgdbm-compat4 arm64 1.23-1 [6294 B]
#0 4.723 Get:5 http://ports.ubuntu.com/ubuntu-ports jammy-updates/main arm64 libperl5.34 arm64 5.34.0-3ubuntu1.1 [4723 kB]
#0 5.146 Get:6 http://ports.ubuntu.com/ubuntu-ports jammy-updates/main arm64 perl arm64 5.34.0-3ubuntu1.1 [232 kB]
#0 5.154 Get:7 http://ports.ubuntu.com/ubuntu-ports jammy-updates/main arm64 libglib2.0-0 arm64 2.72.4-0ubuntu2 [1430 kB]
#0 5.283 Get:8 http://ports.ubuntu.com/ubuntu-ports jammy-updates/main arm64 libglib2.0-data all 2.72.4-0ubuntu2 [4968 B]
#0 5.283 Get:9 http://ports.ubuntu.com/ubuntu-ports jammy/main arm64 libicu70 arm64 70.1-2 [10.5 MB]
#0 6.114 Get:10 http://ports.ubuntu.com/ubuntu-ports jammy-updates/main arm64 libxml2 arm64 2.9.13+dfsg-1ubuntu0.3 [728 kB]
#0 6.185 Get:11 http://ports.ubuntu.com/ubuntu-ports jammy/main arm64 netbase all 6.3 [12.9 kB]
#0 6.300 Get:12 http://ports.ubuntu.com/ubuntu-ports jammy/main arm64 shared-mime-info arm64 2.1-2 [453 kB]
#0 6.494 Get:13 http://ports.ubuntu.com/ubuntu-ports jammy/main arm64 xdg-user-dirs arm64 0.17-2ubuntu4 [53.5 kB]
#0 6.495 Get:14 http://ports.ubuntu.com/ubuntu-ports jammy/main arm64 xz-utils arm64 5.2.5-2ubuntu1 [84.4 kB]
#0 6.500 Get:15 http://ports.ubuntu.com/ubuntu-ports jammy/main arm64 bzip2 arm64 1.0.8-5build1 [34.6 kB]
#0 6.502 Get:16 http://ports.ubuntu.com/ubuntu-ports jammy-updates/main arm64 libdpkg-perl all 1.21.1ubuntu2.2 [237 kB]
#0 6.520 Get:17 http://ports.ubuntu.com/ubuntu-ports jammy/main arm64 libfile-fcntllock-perl arm64 0.22-3build7 [33.7 kB]
#0 6.523 Get:18 http://ports.ubuntu.com/ubuntu-ports jammy/main arm64 pkg-config arm64 0.29.2-1ubuntu3 [47.4 kB]
#0 6.628 debconf: delaying package configuration, since apt-utils is not installed
#0 6.644 Fetched 21.6 MB in 2s (9286 kB/s)
#0 6.655 Selecting previously unselected package liblocale-gettext-perl.
(Reading database ... 4389 files and directories currently installed.)
#0 6.658 Preparing to unpack .../00-liblocale-gettext-perl_1.07-4build3_arm64.deb ...
#0 6.659 Unpacking liblocale-gettext-perl (1.07-4build3) ...
#0 6.672 Selecting previously unselected package perl-modules-5.34.
#0 6.672 Preparing to unpack .../01-perl-modules-5.34_5.34.0-3ubuntu1.1_all.deb ...
#0 6.673 Unpacking perl-modules-5.34 (5.34.0-3ubuntu1.1) ...
#0 6.820 Selecting previously unselected package libgdbm6:arm64.
#0 6.821 Preparing to unpack .../02-libgdbm6_1.23-1_arm64.deb ...
#0 6.823 Unpacking libgdbm6:arm64 (1.23-1) ...
#0 6.834 Selecting previously unselected package libgdbm-compat4:arm64.
#0 6.834 Preparing to unpack .../03-libgdbm-compat4_1.23-1_arm64.deb ...
#0 6.835 Unpacking libgdbm-compat4:arm64 (1.23-1) ...
#0 6.846 Selecting previously unselected package libperl5.34:arm64.
#0 6.847 Preparing to unpack .../04-libperl5.34_5.34.0-3ubuntu1.1_arm64.deb ...
#0 6.847 Unpacking libperl5.34:arm64 (5.34.0-3ubuntu1.1) ...
#0 6.979 Selecting previously unselected package perl.
#0 6.979 Preparing to unpack .../05-perl_5.34.0-3ubuntu1.1_arm64.deb ...
#0 6.984 Unpacking perl (5.34.0-3ubuntu1.1) ...
#0 7.000 Selecting previously unselected package libglib2.0-0:arm64.
#0 7.001 Preparing to unpack .../06-libglib2.0-0_2.72.4-0ubuntu2_arm64.deb ...
#0 7.002 Unpacking libglib2.0-0:arm64 (2.72.4-0ubuntu2) ...
#0 7.027 Selecting previously unselected package libglib2.0-data.
#0 7.027 Preparing to unpack .../07-libglib2.0-data_2.72.4-0ubuntu2_all.deb ...
#0 7.028 Unpacking libglib2.0-data (2.72.4-0ubuntu2) ...
#0 7.038 Selecting previously unselected package libicu70:arm64.
#0 7.039 Preparing to unpack .../08-libicu70_70.1-2_arm64.deb ...
#0 7.039 Unpacking libicu70:arm64 (70.1-2) ...
#0 7.164 Selecting previously unselected package libxml2:arm64.
#0 7.164 Preparing to unpack .../09-libxml2_2.9.13+dfsg-1ubuntu0.3_arm64.deb ...
#0 7.165 Unpacking libxml2:arm64 (2.9.13+dfsg-1ubuntu0.3) ...
#0 7.182 Selecting previously unselected package netbase.
#0 7.183 Preparing to unpack .../10-netbase_6.3_all.deb ...
#0 7.184 Unpacking netbase (6.3) ...
#0 7.195 Selecting previously unselected package shared-mime-info.
#0 7.196 Preparing to unpack .../11-shared-mime-info_2.1-2_arm64.deb ...
#0 7.197 Unpacking shared-mime-info (2.1-2) ...
#0 7.218 Selecting previously unselected package xdg-user-dirs.
#0 7.219 Preparing to unpack .../12-xdg-user-dirs_0.17-2ubuntu4_arm64.deb ...
#0 7.222 Unpacking xdg-user-dirs (0.17-2ubuntu4) ...
#0 7.247 Selecting previously unselected package xz-utils.
#0 7.248 Preparing to unpack .../13-xz-utils_5.2.5-2ubuntu1_arm64.deb ...
#0 7.248 Unpacking xz-utils (5.2.5-2ubuntu1) ...
#0 7.260 Selecting previously unselected package bzip2.
#0 7.260 Preparing to unpack .../14-bzip2_1.0.8-5build1_arm64.deb ...
#0 7.261 Unpacking bzip2 (1.0.8-5build1) ...
#0 7.272 Selecting previously unselected package libdpkg-perl.
#0 7.272 Preparing to unpack .../15-libdpkg-perl_1.21.1ubuntu2.2_all.deb ...
#0 7.273 Unpacking libdpkg-perl (1.21.1ubuntu2.2) ...
#0 7.292 Selecting previously unselected package libfile-fcntllock-perl.
#0 7.292 Preparing to unpack .../16-libfile-fcntllock-perl_0.22-3build7_arm64.deb ...
#0 7.293 Unpacking libfile-fcntllock-perl (0.22-3build7) ...
#0 7.306 Selecting previously unselected package pkg-config.
#0 7.307 Preparing to unpack .../17-pkg-config_0.29.2-1ubuntu3_arm64.deb ...
#0 7.307 Unpacking pkg-config (0.29.2-1ubuntu3) ...
#0 7.322 Setting up xdg-user-dirs (0.17-2ubuntu4) ...
#0 7.329 Setting up libglib2.0-0:arm64 (2.72.4-0ubuntu2) ...
#0 7.335 No schema files found: doing nothing.
#0 7.339 Setting up perl-modules-5.34 (5.34.0-3ubuntu1.1) ...
#0 7.342 Setting up bzip2 (1.0.8-5build1) ...
#0 7.344 Setting up libglib2.0-data (2.72.4-0ubuntu2) ...
#0 7.348 Setting up xz-utils (5.2.5-2ubuntu1) ...
#0 7.352 update-alternatives: using /usr/bin/xz to provide /usr/bin/lzma (lzma) in auto mode
#0 7.352 update-alternatives: warning: skip creation of /usr/share/man/man1/lzma.1.gz because associated file /usr/share/man/man1/xz.1.gz (of link group lzma) doesn't exist
#0 7.352 update-alternatives: warning: skip creation of /usr/share/man/man1/unlzma.1.gz because associated file /usr/share/man/man1/unxz.1.gz (of link group lzma) doesn't exist
#0 7.352 update-alternatives: warning: skip creation of /usr/share/man/man1/lzcat.1.gz because associated file /usr/share/man/man1/xzcat.1.gz (of link group lzma) doesn't exist
#0 7.352 update-alternatives: warning: skip creation of /usr/share/man/man1/lzmore.1.gz because associated file /usr/share/man/man1/xzmore.1.gz (of link group lzma) doesn't exist
#0 7.352 update-alternatives: warning: skip creation of /usr/share/man/man1/lzless.1.gz because associated file /usr/share/man/man1/xzless.1.gz (of link group lzma) doesn't exist
#0 7.352 update-alternatives: warning: skip creation of /usr/share/man/man1/lzdiff.1.gz because associated file /usr/share/man/man1/xzdiff.1.gz (of link group lzma) doesn't exist
#0 7.352 update-alternatives: warning: skip creation of /usr/share/man/man1/lzcmp.1.gz because associated file /usr/share/man/man1/xzcmp.1.gz (of link group lzma) doesn't exist
#0 7.352 update-alternatives: warning: skip creation of /usr/share/man/man1/lzgrep.1.gz because associated file /usr/share/man/man1/xzgrep.1.gz (of link group lzma) doesn't exist
#0 7.352 update-alternatives: warning: skip creation of /usr/share/man/man1/lzegrep.1.gz because associated file /usr/share/man/man1/xzegrep.1.gz (of link group lzma) doesn't exist
#0 7.352 update-alternatives: warning: skip creation of /usr/share/man/man1/lzfgrep.1.gz because associated file /usr/share/man/man1/xzfgrep.1.gz (of link group lzma) doesn't exist
#0 7.354 Setting up netbase (6.3) ...
#0 7.362 Setting up liblocale-gettext-perl (1.07-4build3) ...
#0 7.365 Setting up libgdbm6:arm64 (1.23-1) ...
#0 7.367 Setting up libicu70:arm64 (70.1-2) ...
#0 7.370 Setting up libgdbm-compat4:arm64 (1.23-1) ...
#0 7.372 Setting up libxml2:arm64 (2.9.13+dfsg-1ubuntu0.3) ...
#0 7.375 Setting up libperl5.34:arm64 (5.34.0-3ubuntu1.1) ...
#0 7.379 Setting up perl (5.34.0-3ubuntu1.1) ...
#0 7.386 Setting up shared-mime-info (2.1-2) ...
#0 7.987 Setting up libdpkg-perl (1.21.1ubuntu2.2) ...
#0 7.989 Setting up libfile-fcntllock-perl (0.22-3build7) ...
#0 7.992 Setting up pkg-config (0.29.2-1ubuntu3) ...
#0 8.018 Processing triggers for libc-bin (2.35-0ubuntu3.1) ...
#0 8.048 ./setup.sh: line 22: curl: command not found
#0 8.049 ./setup.sh: line 24: rustup: command not found
#0 8.049 ./setup.sh: line 25: rustup: command not found
#0 8.050 ./setup.sh: line 26: rustup: command not found
#0 8.051 ./setup.sh: line 27: rustup: command not found
#0 8.051 ./setup.sh: line 28: rustup: command not found
#0 8.052 ./setup.sh: line 30: cargo: command not found
#0 8.053 ./setup.sh: line 31: cargo: command not found
#0 8.053 ./setup.sh: line 32: cargo: command not found
#0 8.054 ./setup.sh: line 33: rustup: command not found
#0 8.055 ./setup.sh: line 36: git: command not found
------
Dockerfile-with-features:11
--------------------
   9 |     ENV PATH="/root/.cargo/bin:$PATH"
  10 |     
  11 | >>> RUN bash ./setup.sh
  12 |     
  13 |     FROM $_DEV_CONTAINERS_BASE_IMAGE AS dev_containers_target_stage
--------------------
ERROR: failed to solve: process "/bin/sh -c bash ./setup.sh" did not complete successfully: exit code: 127
[9329 ms] Error: Command failed: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /var/folders/46/_3zl9m6s6jv1qv13s33nj1j00000gn/T/devcontainercli/container-features/0.40.0-1685449859513/Dockerfile-with-features -t vsc-spiderlightning-30b33caf300869ef9610e9f16aa57f25d39de8b7911bc68af10116184f397fd5 --target dev_containers_target_stage --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label /Users/christianweyer/Sources/spiderlightning/.devcontainer
[9329 ms]     at Lse (/Users/christianweyer/.vscode/extensions/ms-vscode-remote.remote-containers-0.292.0/dist/spec-node/devContainersSpecCLI.js:1947:1698)
[9329 ms]     at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
[9329 ms]     at async wD (/Users/christianweyer/.vscode/extensions/ms-vscode-remote.remote-containers-0.292.0/dist/spec-node/devContainersSpecCLI.js:1946:3889)
[9329 ms]     at async rO (/Users/christianweyer/.vscode/extensions/ms-vscode-remote.remote-containers-0.292.0/dist/spec-node/devContainersSpecCLI.js:1946:2808)
[9329 ms]     at async Xse (/Users/christianweyer/.vscode/extensions/ms-vscode-remote.remote-containers-0.292.0/dist/spec-node/devContainersSpecCLI.js:1961:2626)
[9329 ms]     at async vh (/Users/christianweyer/.vscode/extensions/ms-vscode-remote.remote-containers-0.292.0/dist/spec-node/devContainersSpecCLI.js:1961:3741)
[9329 ms]     at async gae (/Users/christianweyer/.vscode/extensions/ms-vscode-remote.remote-containers-0.292.0/dist/spec-node/devContainersSpecCLI.js:2092:10213)
[9329 ms]     at async mae (/Users/christianweyer/.vscode/extensions/ms-vscode-remote.remote-containers-0.292.0/dist/spec-node/devContainersSpecCLI.js:2092:9954)
[9332 ms] Exit code 1
[9333 ms] Command failed: /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin) --ms-enable-electron-run-as-node /Users/christianweyer/.vscode/extensions/ms-vscode-remote.remote-containers-0.292.0/dist/spec-node/devContainersSpecCLI.js up --user-data-folder /Users/christianweyer/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-containers/data --container-session-data-folder /tmp/devcontainers-2ccd0251-3333-42c4-b712-4a6ab0c9dc391685449858170 --workspace-folder /Users/christianweyer/Sources/spiderlightning --workspace-mount-consistency cached --id-label devcontainer.local_folder=/Users/christianweyer/Sources/spiderlightning --id-label devcontainer.config_file=/Users/christianweyer/Sources/spiderlightning/.devcontainer/devcontainer.json --log-level debug --log-format json --config /Users/christianweyer/Sources/spiderlightning/.devcontainer/devcontainer.json --default-user-env-probe loginInteractiveShell --mount type=volume,source=vscode,target=/vscode,external=true --skip-post-create --update-remote-user-uid-default on --mount-workspace-git-root true
[9334 ms] Exit code 1
```

**To Reproduce**
Choose `'Dev Containers: Open Folder in Container'` in VS Code.

The above error occurs in the terminal.

**Additional context**
Running on macOS 13.3.1a on M1 Max

**Comment by danbugs** [comment]
I'm looking into this.

**Comment by danbugs** [comment]
@ChristianWeyer ~ I successfully replicated this bug and fixed it w/ PR #403 . Please, let me know if you run into any other issues.

**Comment by ChristianWeyer** [comment]
Perfect, works now.
Thanks!
